### PR TITLE
Login page Test

### PR DESCRIPTION
**Pull request by alszla**

### CODE REVIEW

File: src/main/java/core/utils/PropertiesReader.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+            System.setProperty("BROWSER", properties.getProperty("BROWSER"));
+            String browser = System.getProperty("BROWSER");
+            System.setProperty(browser.concat(".implicitWaitTime"),
+                    properties.getProperty(browser.concat(".implicitWaitTime")));
+            System.setProperty(browser.concat(".explicitWaitTime"),
+                    properties.getProperty(browser.concat(".explicitWaitTime")));
+        } catch (IOException ioex) {
+            // TODO Replace with logger.
+            System.out.println("Input file path invalid!");
```
Comment (Cxristian) [inline src/main/java/core/utils/PropertiesReader.java]: i think that do you need logs..

File: src/main/java/oshpark/ui/pages/Home.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    private WebElement userPanel;
+
+    /**
+     * Homepage class constructor.
+     *
+     * @param driver a WebDriver object.
+     */
+    public Home(WebDriver driver) {
+        super(driver);
+        uri = "https://oshpark.com/";
```
Comment (Cxristian) [inline src/main/java/oshpark/ui/pages/Home.java]: your url should be read from properties

File: src/main/java/oshpark/ui/pages/Home.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        PageFactory.initElements(driver, this);
```
Comment (Cxristian) [inline src/main/java/oshpark/ui/pages/Home.java]: you should start your page factory on your base page in order for them to inherit all classes that extend from base page

File: src/test/java/LoginTest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * Test class for Login page.
+ *
+ * @author Alejandro Sánchez Luizaga
+ * @version 1.0
+ */
+public class LoginTest {
```
Comment (Cxristian) [inline src/test/java/LoginTest.java]: I don't understand why you need your Logintest class.

File: gradle.properties [hunk 5]
```diff
@@ -0,0 +1,7 @@
+BROWSER=Chrome
+
+Chrome.implicitWaitTime = 30
+Chrome.explicitWaitTime = 40
+
```
[the remark this answers is not in the document]
Comment (alszla) [reply gradle.properties]: Well, they are different considering that Chrome works a little bit faster than Firefox.

File: src/main/java/core/utils/PropertiesReader.java (hunk 1 above)
Comment (alszla) [reply src/main/java/core/utils/PropertiesReader.java]: Right. I'll be fixing this as soon as possible.

File: src/main/java/oshpark/ui/pages/Home.java (hunk 3 above)
Comment (alszla) [reply src/main/java/oshpark/ui/pages/Home.java]: Thanks. I didn't know this. 
Somehow I thought that elements should be initated once they were defined by @FindBy annotation in each class.
I'll be following your advice. PageFactory is going to be moved to BasePage.

File: src/test/java/LoginTest.java (hunk 4 above)
Comment (alszla) [reply src/test/java/LoginTest.java]: That's right. It is not needed anymore. But this file is a remnant now. I mean, I started working the TestNG assertion even before writing the scenario as we did with the calculadora.org page.

File: src/main/java/oshpark/ui/pages/Home.java (hunk 3 above)
Comment (alszla) [reply src/main/java/oshpark/ui/pages/Home.java]: Done

File: src/main/java/core/utils/PropertiesReader.java (hunk 1 above)
Comment (alszla) [reply src/main/java/core/utils/PropertiesReader.java]: Done

File: src/main/java/oshpark/ui/pages/Home.java (hunk 2 above)
Comment (alszla) [reply src/main/java/oshpark/ui/pages/Home.java]: Right. This url will be read from a properties file.

**Pull request merged**